requirements/docs.in

```diff
@@ -6,7 +6,7 @@ ipywidgets
 packaging
 requests
 sphinx
-sphinx-autodoc-typehints>=1.17.0
+sphinx-autodoc-typehints==1.17.0  # later incompatible with with book-theme
```

Comment:
Issue reference?

Reply:
What do you mean? I am not aware of any issues. `book-theme` has an upper limit on `sphinx-autodoc-typehints`.

Reply:
So why are we freezing it then, if there is no issue?

Reply:
Because of how pip resolves dependencies. It was unable to find compatible versions without the pin. It tried to use the latest `sphinx-autodoc-typehints` and complained about a clash with `sphinx-book-theme`.

Reply:
Does it work if we swap the order of the two lines?

Reply:
It does not.
The problem is not actually with `sphinx-autodoc-typehints` directly but with its dependency on sphinx:
```
  Sphinx>=5.1.1 (from sphinx-autodoc-typehints==1.19.2->-r requirements/docs.in (line 9))
  sphinx<5,>=3 (from sphinx-book-theme==0.3.3->-r requirements/docs.in (line 10))
```
Maybe we can solve this by pinning `sphinx` instead. But that seems more heavy handed than making the requirement on  `sphinx-autodoc-typehints` stricter.